src/openagi/actions/tools/exasearch.py

```diff
@@ -1,30 +1,75 @@
 from openagi.actions.base import BaseAction
-import os
 from pydantic import Field
 from openagi.exception import OpenAGIException
+from typing import ClassVar, Dict, Any
+import os
+import warnings
 
 try:
-   from exa_py import Exa
+    from exa_py import Exa
 except ImportError:
-  raise OpenAGIException("Install Exa Py with cmd `pip install exa_py`")
+    raise OpenAGIException("Install Exa Py with cmd `pip install exa_py`")
 
-class ExaSearch(BaseAction):
+class ConfigurableAction(BaseAction):
```

Comment:
@tarun-aiplanet why are we redefining the ConfigurableAction class?

Reply:
we need this for configuring the api keys and other user defined params right?